sled-storage/src/manager.rs

```diff
@@ -539,21 +581,37 @@ impl StorageManager {
             }
         }
 
+        // We want to add all U.2 disks concurrently rather than sequentially.
```

Comment:
IMO it's a little arbitrary to be doing this for U.2s, and not M.2s, right?

I think I see that we made this choice because these functions act on `&mut self`? (If I'm wrong, please correct me) and that would make parallelized calls not possible

But I wonder if this is worth a slight refactor, to split:

- `add_disk`, which is `&mut self`, and currently:
  - `Disk::new` +
  - Update "self" (either via `self.resources`, queues, state, etc)

Into something like:

- (&mut self) "Check we're ready for this disk, or enqueue it for later if not"
- (&self) "Actually call `Disk::new`"
- (&mut self) "Update state" (call `insert_disks`, update `self.queued_drives`, `self.state`, etc)

This would let us keep the calls to `Disk::new` parallel, while avoiding some of the splayed state management